src/Tribe/Aggregator/Record/Queue_Realtime.php

```diff
@@ -110,35 +111,48 @@ public function render_update_message() {
 	}
 
 	/**
-	 * Handle queue ajax requests
+     * Action to reply every time a heart beat is executed to send the progress of EA if an EA record is present.
+     *
+	 * @param array<string, mixed> $response The current response object
+	 * @param array<string, mixed> $data An array with the data from the client.
+	 *
+	 * @return array<string, mixed> An array used to construct the heart beat response.
 	 */
-	public function ajax() {
-		$this->record_id = (int) $_POST['record'];
+	public function receive_heartbeat( array $response, array $data ) {
+		if ( empty( $data['ea_record'] ) ) {
+			return $response;
+		}
 
-		// Nonce check
-		$this->ajax_operations->verify_or_exit( $_POST['check'], $this->get_ajax_nonce_action(), $this->get_unable_to_continue_processing_data() );
+		$this->record_id = absint( $data['ea_record'] );
 
-		// Load the queue
-		/** @var \Tribe__Events__Aggregator__Record__Queue_Interface $queue */
-		$queue = $this->queue ? $this->queue : Tribe__Events__Aggregator__Record__Queue_Processor::build_queue( $this->record_id );
-		// We always need to setup the Current Queue
-		$this->queue_processor->set_current_queue( $queue );
+		if ( $this->record_id === 0 ) {
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 7ccc1abb8d6e0c4cf636dbc6a9f3e0ad</sub>